synapseclient/api/table_services.py

```diff
@@ -43,6 +45,54 @@ class ViewTypeMask(int, Enum):
     MATERIALIZED_VIEW = 0x200
 
 
+async def create_table_snapshot(
+    table_id: str,
+    comment: Optional[str] = None,
+    label: Optional[str] = None,
+    activity_id: Optional[str] = None,
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Union[str, int]]:
+    """
+    Creates a table snapshot using the Synapse REST API.
+
+    Arguments:
+        table_id: Table ID to create a snapshot for.
+        comment: Optional snapshot comment.
+        label: Optional snapshot label.
+        activity_id: Optional activity ID or activity instance applied to snapshot version.
+        synapse_client: If not passed in and caching was not disabled by
+            `Synapse.allow_client_caching(False)` this will use the last created
+            instance from the Synapse class constructor.
+
+    Returns:
+        A dictionary containing the snapshot response.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+
+    if activity_id and not isinstance(activity_id, str):
+        activity_id = str(activity_id)
+
```

Comment:
Are you just being extra careful here? `activity_id` can only be of type `str` or `None`.